### ISSUE TITLE
x64 Realtime Stacks not working after switch to RuntimeInformation.OSArchitecture

### ISSUE THREAD

**Issue by Alois-xx** [microsoft/perfview#478]
Commit f3f49e9f contains the switch to System.Runtime.InteropServices.RuntimeInformation, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a. That version breaks x64 Realtime stackwalks because RuntimeInformation.OSArchitecture returns x86 on x64 which is clearly an error.

The version is supplied with .NET 4.7.1 which forwards the type to mscorlib where it is defined as 
```
public static Architecture OSArchitecture
{
	[SecuritySafeCritical]
	get
	{
		object obj = RuntimeInformation.s_osLock;
		lock (obj)
		{
			if (!RuntimeInformation.s_osArch.HasValue)
			{
				Win32Native.SYSTEM_INFO sYSTEM_INFO;
				Win32Native.GetNativeSystemInfo(out sYSTEM_INFO);
				RuntimeInformation.s_osArch = new Architecture?(RuntimeInformation.GetArchitecture(sYSTEM_INFO.dwOemId));
			}
		}
		return RuntimeInformation.s_osArch.Value;
	}
}
```
That has several problems. 

1. dwOemId is at least on Win10 the wrong property of the struct to determine the OS architecture 
   The code in RuntimeInformation.GetArchitecture wrongly shifts it by 16 bits which results always in 0.
  
    ushort num = (ushort)(sysinfo.dwOemId >> 16);
   That must be fixed inside mscorlib inside .NET 4.7.1.

2. The supplied Nuget package for System.Runtime.InteropServices.RuntimeInformation 4.3 supplies an assembly for users who do not have .NET 4.7.1 installed but that version relies on API sets 

```
[DllImport("api-ms-win-core-sysinfo-l1-2-0.dll")]
internal static extern void GetNativeSystemInfo(out Interop.mincore.SYSTEM_INFO lpSystemInfo);
```
 which breaks Windows 7 users. API Sets are a feature introduced with Windows 8.

Point 2 seems to be already addressed with 4.3.0-preview1-24530-04 but that will still fail on all machines where .NET 4.7.1 is installed. 

Are there plans to skip support for Win 7 with all of these changes? If yes I would need to live with a fork before all the .NET Core changes were added which seem to break quite a few cases. It is unfortunate that the Nuget package has the same assembly version than the one from .NET 4.7.1 which forces the usage of the buggy .NET 4.7.1 System.RuntimeInformation.dll from the GAC (4.0.1.0).

**Comment by Alois-xx** [comment]
Any update on this one? I would expect a solution that also works on a .NET 4.7.1 installation with no hotfixes. Otherwise tools will break. TraceEvent should be able to work on pretty much any .NET version.

**Comment by vancem** [comment]
Given that the underlying APIs are broken on some versions avoiding the API is best.  In this case it happened to be easy (it was called only to determine the current pointer size, which we can get using IntPtr.Size.  

On a related note.  Has any bug been made to actually fix the API in .NET.   PerfView no longer needs it, but we still need to fix it.

**Comment by Alois-xx** [comment]
IntPtr.Size will not do it because if you are running on a 64 bit OS you will wrongly detect that the ETL pointer size is 4 if you are running a 32 bit application.

**Comment by Alois-xx** [comment]
I have logged request to check for desktop clr https://github.com/dotnet/corefx/issues/27386

**Comment by vancem** [comment]
Sigh,  I was not thinking straight.   I was thinking in real time it matches the process, but you are right, it does not.     I am going to abandon this PR

**Comment by sharwell** [comment]
@vancem Did you mean to close the pull request or the issue?

**Comment by brianrob** [comment]
Fixed via #576.
